### DISCUSSION TITLE
Parkour challenge levels

### DISCUSSION THREAD

**Discussion by no-lex** [redeclipse/base#1164] [Maps and Content Showcase]
The recent addition of parkour challenges as a game type has allowed RE2 to have a singleplayer aspect, and allow for a more guided learning experience of RE's advanced parkour features.

I have ported the first four levels of Port to 1.9.9 (challenges-port-0x) as the first four challenges in this gamemode, but having twenty or more levels would be great for allowing a decent amount of singleplayer gameplay without undue effort required.

Possible ideas for challenges:
- Tutorial set of levels to introduce parkour concepts
- Sets of levels with consistent themes
- Ports of levels (crueldozen, remaining ports)
- Difficult sections of existing maps

If any community members would like to contribute challenge maps for RE 2.0 release, that would be a relatively simple and easy way (vs. making an FFA map or a full race map e.g.) to help contribute to the project.

Criteria for submissions:
- Fully textured and lit
- Appropriate mapauthor and mapdesc strings; textures culled and calclight'd
- Three required entities: a single spawn point and a start/finish checkpoint
- No more than 1 spawn point and no intermediate checkpoints
- No more than 10 seconds to complete by a skilled runner (challenges should be short)
- Deathmat needs a reason to exist (lasers, water, etc.); difficult to parkour geometry recommended instead of large deathmat areas if possible
- Only one obstacle in the level (not a series of obstacles like a full length race map)
- No route markers required; map should not need them anyway (there should be one obvious "right" way)
- Individual contributions must match the set they belong to (e.g. all tuto levels should share a theme)

For examples, look at the four levels included which are a "matched set". I will be adding more maps in the future that are part of the set, as ports of Temka's _port_ map.

**Comment by lambda-11235** [comment]
I call wall jump thingy.
![20190708010353](https://user-images.githubusercontent.com/16354815/60793363-6c524d00-a11c-11e9-9e1b-c5f84b8e64f2.png)

**Comment by no-lex** [comment]
Tutorial level? If it's going after the port sequence of levels it should be more difficult than port-20, but a cursory glance makes this appear more suitable as a tutorial level (does not seem difficult). For tutorial levels, having programmed `echo` responses explaining how to do things (ala _escapeultima_) is something that would help make explainations easier and clearer than putting text on the walls.

**Comment by lambda-11235** [comment]
Ok, I'll change it. I was thinking of it going after port 4, but I'll try something else.

**Comment by SniperGoth** [comment]
Is it planned to be separated by diffculty levels?
I could cut sections from some of my harder/extreme ones to be added then.
Like this one part from "euphoria"  for example : 
![20190708143508](https://user-images.githubusercontent.com/24359960/60830460-a7409900-a18d-11e9-9846-0e1f96928a55.png)

**Comment by no-lex** [comment]
The plan is to have levels in sequential order by difficulty; port is the first set of "real" levels (which are no longer particularly difficult in 2.0) and more difficult sets of levels can follow (e.g. what you propose above, or a port of the _crueldozen_ set, etc.) Theme can be different than _port_ as long as it's part of a consistent set.

@lambda-11235 yeah, the port sequence of levels is planned to be a contiguous block of challenges, so either making a tutorial for wall climbing (perhaps easier and without death water?) or making a more challenging level would work. I'd like not to have a one-off level however, and would prefer something as part of a set (like tuto level set or port)

**Comment by lambda-11235** [comment]
Current work on port 19, non-textured.

![20190710184015](https://user-images.githubusercontent.com/16354815/61015839-3ee6e880-a342-11e9-98df-39ab67ed9727.png)

v0.1:
[port19-lambda.zip](https://github.com/red-eclipse/world/files/3378773/port19-lambda.zip)

v0.2, replace waterfalls with non-parkourable geometry:
[port19-lambda.zip](https://github.com/red-eclipse/world/files/3379614/port19-lambda.zip)

v0.3, add supports:
[port19-lambda.zip](https://github.com/red-eclipse/world/files/3379879/port19-lambda.zip)

v0.4, add textures:
[port19-lambda.zip](https://github.com/red-eclipse/world/files/3379912/port19-lambda.zip)

**Comment by lambda-11235** [comment]
I've started an early version of the trellis challenge. Please tell me what you think of its difficulty.
![20190714152027](https://user-images.githubusercontent.com/16354815/61190064-2c72f480-a64b-11e9-9fc2-079e9f13dea3.png)

v0.1: 
[trellis.zip](https://github.com/red-eclipse/world/files/3390477/trellis.zip)
